### ISSUE TITLE
Python 2.7 compatibility

### ISSUE THREAD

**Issue by tbrx** [probtorch/pytorch#23]
I believe we should still be supporting python 2.7 with the distributions code (as well as 3.x). The pre-existing `distributions-api` branch includes some code which doesn't parse on 2.7. 

My understanding is that pytorch in general targets 2.7 and 3.5+ — is this correct?

**Comment by neerajprad** [comment]
That is correct - PyTorch (and Pyro, as well) supports both 2.7 and 3.5+. We used the [six](https://pypi.python.org/pypi/six) library to handle imports of common library functions. PyTorch seems to have an internal `_six` file to specify the appropriate python 2/3 packages to bind to the namespace depending on the runtime python version.
